WalletWasabi/Services/SingleInstanceChecker.cs

```diff
@@ -0,0 +1,58 @@
+using NBitcoin;
+using Nito.AsyncEx;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace WalletWasabi.Services
+{
+	public class SingleInstanceChecker : IDisposable
+	{
+		private const string MutexString = "WalletWasabiSingleInstance";
+		private bool _disposedValue;
+
+		public SingleInstanceChecker(Network network)
+		{
+			Network = network;
+		}
+
+		private IDisposable SingleApplicationLockHolder { get; set; }
+		private Network Network { get; }
+
+		public async Task CheckAsync()
+		{
+			using CancellationTokenSource cts = new CancellationTokenSource(TimeSpan.Zero);
```

Comment:
There was a time when Microsoft published the .net framework design guidelines and the recomendation was always implement the IDisposable pattern completed and check if it was already disposed in order to prevent the creation of unmanaged resources again.

In this case I can call this method after disposing the instances. I think we should prevent such situations.

```suggestion
			if (_disposedValue)
			{
				throw new ObjectDisposedException(nameof(SingleInstanceChecker));
			} 
```

Reply:
Totally makes sense in this scenario.